backend/export.py

```diff
@@ -59,10 +72,45 @@ def fetch_tree_messages(
         qry = qry.filter(not_(Message.review_result), Message.review_count > 2)
     elif review_result is True:
         qry = qry.filter(Message.review_result)
+    if limit is not None and limit > 0:
+        qry = qry.limit(limit)
 
     return qry.all()
 
 
+def get_events_for_messages(db: Session, message_ids: list[UUID]) -> dict[UUID, ExportMessageEvent]:
+    events = {}
+    emojis = db.query(MessageEmoji).filter(MessageEmoji.message_id.in_(message_ids)).all()
```

Comment:
(We have to test this with the the production db size, I had similar code before for labels which I re-wrote after the query took > 0.5h to execute for 100k messages .. i.e. `fetch_tree_messages_and_avg_labels` function does the join directly .. but needs to be tried)

Reply:
fair point. let's test and see